packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -930,13 +933,16 @@ class CupertinoSliverNavigationBar extends StatefulWidget {
     this.heroTag = _defaultHeroTag,
     this.stretch = false,
     this.bottomMode = NavigationBarBottomMode.automatic,
+    this.onSearchActiveChanged,
```

Comment:
Is this needed with the search field exposed? Could users just listen to changes directly by providing their own search field?

Reply:
The way to show a search view right now without this callback would be to monitor if the search field has been focused by attaching a focus node to the search field. But there might be a situation where the search view should remain open even when the search field is no longer focused (e.g a search has completed with some results, and the results are focused).